rust/src/traits/samplers/uniform/mod.rs

```diff
@@ -48,11 +32,25 @@ pub trait SampleUniform: Sized {
     fn sample_standard_uniform(constant_time: bool) -> Fallible<Self>;
 }
 
+
 impl<T, B> SampleUniform for T
     where 
         T: SampleMantissa<Bits=B>,
         B: ExactIntCast<usize> + Sub<Output=B> + One,
         usize: ExactIntCast<B> {
+
+    /// This algorithm is taken from [Mironov (2012)](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.366.5957&rep=rep1&type=pdf)
+    /// and is important for making some of the guarantees in the paper.
+    ///
+    /// The idea behind the uniform sampling is to first sample a "precision band".
+    /// Each band is a range of floating point numbers with the same level of arithmetic precision
+    /// and is situated between powers of two.
+    /// A band is sampled with probability relative to the unit of least precision using the Geometric distribution.
+    /// That is, the uniform sampler will generate the band [1/2,1) with probability 1/2, [1/4,1/2) with probability 1/4,
+    /// and so on.
+    ///
+    /// Once the precision band has been selected, floating numbers numbers are generated uniformly within the band
+    /// by generating a 52-bit mantissa uniformly at random.
```

Comment:
Not sure if anything can be done about this, but this block doesn't seem to appear anywhere in the rustdoc.

Reply:
I moved it up to the trait impl. Good to keep in mind in the future.